pkg/controller/jobframework/validation.go

```diff
@@ -120,7 +120,7 @@ func ValidateQueueName(obj client.Object) field.ErrorList {
 	return allErrs
 }
 
-func validateUpdateForQueueName(oldJob, newJob GenericJob, defaultQueueExist func(string) bool) field.ErrorList {
+func ValidateUpdateForQueueName(oldJob, newJob GenericJob, defaultQueueExist func(string) bool) field.ErrorList {
```

Comment:
Is this exporting needed? I would prefer to minimize the production code changes. It is probably ok, but note that the `jobframework` package is meant to be used by users at compile time. So, breaking any exported function in the future can cause issues on upgrades for users.

Reply:
Reverted.